lib/decompress/zstd_decompress.c

```diff
@@ -1656,14 +1656,16 @@ size_t ZSTD_initDStream_usingDict(ZSTD_DStream* zds, const void* dict, size_t di
 size_t ZSTD_initDStream(ZSTD_DStream* zds)
 {
     DEBUGLOG(4, "ZSTD_initDStream");
-    return ZSTD_initDStream_usingDDict(zds, NULL);
+    FORWARD_IF_ERROR(ZSTD_DCtx_reset(zds, ZSTD_reset_session_only), "");
+    return ZSTD_DCtx_refDDict(zds, NULL);
```

Comment:
This changes the return value, it should be `ZSTD_startingInputLength(zds->format)`